backend/fms_core/models/derived_sample.py

```diff
@@ -96,21 +111,56 @@ class DerivedSample(TrackedModel):
 
     biosample = models.ForeignKey("Biosample", on_delete=models.PROTECT, related_name="derived_samples",
                                   help_text="Biosample associated to this DerivedSample")
-
     sample_kind = models.ForeignKey(SampleKind, on_delete=models.PROTECT,
-                                    help_text="Biological material collected from study subject "
-                                              "during the conduct of a genomic study project.")
-
-
-    experimental_group = models.JSONField(blank=True, default=list,
-                                          validators=[JsonSchemaValidator(EXPERIMENTAL_GROUP_SCHEMA)],
+                                    help_text="Biological material collected from study subject during the conduct of a genomic study project.")
+    experimental_group = models.JSONField(blank=True, default=list, validators=[JsonSchemaValidator(EXPERIMENTAL_GROUP_SCHEMA)],
                                           help_text="Sample group having some common characteristics. "
                                                     "It is the way to designate a subgroup within a study.")
-
     tissue_source = models.CharField(max_length=200, blank=True, choices=TISSUE_SOURCE_CHOICES,
                                      help_text="Can only be specified if the biospecimen type is DNA or RNA.")
-
     library = models.CharField(max_length=200, blank=True, null=True)
-
     index = models.CharField(max_length=200, blank=True, null=True)
 
+    def extracted_from(self): # returns a tuple of samples (extracted, extracted_from)
```

Comment:
Could you turn this into a '@property'? 

Also Im not sure about the logic. You are using an iterable collection? So if I want to get the extracted_from from a sample, I would get the first and only sample.extracted_from, if it's a pool I get the first item of an iterable collection?

Reply:
you would get (None, None) since pools are not a byproduct of an extraction...

Reply:
referring to my other comments I just posted (below), we could possibly create a type of custom exception for operations wrongly attempted

The return value (None, None) does not point to the problem

Reply:
There is no problem here as far as i can see (at least not in this case) since it is a function to return the extracted_from which it does (and a bit more for practical reasons). This operation is a case where we do not have to worry ... i want the extracted_from a sample its either something or nothing (if the sample is not produced through extraction).